quickstart-templates/msi-as-a-fic-secretless/main.bicep

```diff
@@ -0,0 +1,103 @@
+extension microsoftGraphV1
+
+// TEMPLATE DESCRIPTION
+// Creates a secret-less client application, using a user-assigned managed identity
+// as the credential (configured as part of the application's federated identity credential).
+// The script optionally
+
+@description('Specifies the name of cloud environment to run this deployment in.')
+@allowed(['publicCloud','usGov','china','usNat','usSec'])
+param cloudEnvironment string
+
+// NOTE: Microsoft Graph Bicep file deployment is only supported in Public Cloud
+@description('Audience uris for public and national clouds')
+param audiences object = {
+  publicCloud: {
```

Comment:
Bicep has an `environment()` function and its output contains the name of current cloud https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/bicep-functions-deployment#example-1. Should we use that as the key in this object?

Reply:
Will investigate -  I use it elsewhere - not sure why it didn't occur to me.

Reply:
Difficult to know what the names are for USSec and USNat. Those two are not documented from what I can tell, and using Az CLI/PS only public, china and USGov cloud endpoints show up in azure environment list response.

Would only be able to test this in USSec and USNat in those tenants in that environment.  Will ask some internal folks...

Reply:
@jason-dou best guess based on documentation on Azure Government Clouds, but I don't know the "official" Azure names for USNat and USSec. USNat looks like it's USGov, but with 2 regions exclusively for DoD (which I assume are air-gapped). USSec looks altogether different, but then there's also top secret!  Very confused.  See https://azure.microsoft.com/en-us/blog/azure-government-top-secret-now-generally-available-for-us-national-security-missions/

Reply:
Got confirmation from our contacts on the naming.  Will update

Reply:
Updated after confirmation from the AGC and ARM teams who corroborated the values.